### DISCUSSION TITLE
Downgrade deliverd Libraries

### DISCUSSION THREAD

**Discussion by FabioEasylogix** [wixtoolset/issues#7875] [Questions]
How can we manage to use lower file versions for delivered DLLs? We have changed some dependencies and have the problem that when overwriting existing installations, some DLLs are not copied because all these DLLs have a lower file version than in the last version.
Is there a solution how to force the copying of DLLs in the installation directory?

This only happens when we overwrite an older version. If we run the installer twice, it works with the repair option, but the customer does not recognize the missing DLLs immediately. 

Any idea how to fix this?

**Comment by nirbar** [comment]
See [REINSTALLMODE](https://learn.microsoft.com/en-us/windows/win32/msi/reinstallmode)- it configures the behavior overwrite mode (applied to all the files)
If you want to do that for hand-picked files then you can use my WiX extension [PanelSwWixExtension4](https://www.nuget.org/packages/PanelSwWixExtension4) for WiX4 or [PanelSwWixExtension](https://www.nuget.org/packages/PanelSwWixExtension) for WiX3:

```
<File ...>
   <PanelSW:ForceVersion />
</File>
```

**Comment by FabioEasylogix** [comment]
Thank you for your answer, but we have many different dlls, we have found a workaround using a post process with CompanionFiles, we have one dll with dynamic version number and set all others to KeyPath="yes" and CompanionFile="IDrefDLL".

If there is a better solution, I'm still interesting in some "clean" options to solve this.
